src/AggregateSpecsPlugin.ts

```diff
@@ -86,37 +126,33 @@ const AggregateSpecsPlugin: Plugin = (builder) => {
         id: "min",
         humanLabel: "minimum",
         HumanLabel: "Minimum",
-        isSuitableType: isNumberLike,
+        isSuitableType: isMinMaxable,
         sqlAggregateWrap: (sqlFrag) => sql.fragment`min(${sqlFrag})`,
       },
       {
         id: "max",
         humanLabel: "maximum",
         HumanLabel: "Maximum",
-        isSuitableType: isNumberLike,
+        isSuitableType: isMinMaxable,
         sqlAggregateWrap: (sqlFrag) => sql.fragment`max(${sqlFrag})`,
       },
       {
         id: "average",
         humanLabel: "mean average",
         HumanLabel: "Mean average",
-        isSuitableType: isNumberLike,
+        isSuitableType: isNumberOrDateLike,
         sqlAggregateWrap: (sqlFrag) => sql.fragment`avg(${sqlFrag})`,
 
         // An AVG(...) ends up more precise than any individual value; see
         // https://www.postgresql.org/docs/current/functions-aggregate.html for
         // how the avg aggregate changes result type.
         pgTypeAndModifierModifier: convertWithMapAndFallback(
           {
-            [SMALLINT_OID]: NUMERIC_OID, // smallint -> numeric
-            [INTEGER_OID]: NUMERIC_OID, // integer -> numeric
-            [BIGINT_OID]: NUMERIC_OID, // bigint -> numeric
-            [NUMERIC_OID]: NUMERIC_OID, // numeric -> numeric
```

Comment:
With the numeric fallback these specific conversions weren't necessary.

Reply:
I'd rather they were present though, they make it clear that mapping from smallint to numeric is deliberate (from the PostgreSQL docs) rather than a fallback behaviour that should be overridden. (Effectively they're there so I don't get pull requests suggesting mapping them to other types.) Please restore them.